pilot/pkg/model/service.go

```diff
@@ -813,20 +813,45 @@ func BuildSubsetKey(direction TrafficDirection, subsetName string, hostname Host
 	return fmt.Sprintf("%s|%d|%s|%s", direction, port, subsetName, hostname)
 }
 
+// BuildDNSSrvSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
+// The proxy queries Pilot with this key to obtain the list of instances in a subset.
+// This is used only for the SNI-DNAT router. Do not use for other purposes.
+func BuildDNSSrvSubsetKey(direction TrafficDirection, subsetName string, hostname Hostname, port int) string {
+	return fmt.Sprintf("%s_.%d_.%s_.%s", direction, port, subsetName, hostname)
```

Comment:
Do we really need direction ? Are there 2 directions even possible for DNAT ? 
Also, can we use the more common _ in front of the name (like in dnssd) ?

Having a domain component start with a number is a bit risky - I think it was not allowed in the 
original RFC, was relaxed later, but may still cause some confusion. Having _ in front would avoid this.

Reply:
I see you use direction_ to detect it's of this time - we could still use a prefix, like:
  _istio._8080._v1.products.ns1.svc.cluster.local

Reply:
This name maps with the cluster names. Really don’t want to add another special string there. 

The underscore in the end is dns srv format isn’t it? GRPC and others use the srv format right?

I don’t mind changing it but will have to change the SNI format as well.